validator/src/main/java/org/ethereum/beacon/validator/attester/BeaconChainAttesterImpl.java

```diff
@@ -0,0 +1,127 @@
+import org.ethereum.beacon.core.operations.attestation.AttestationData;
+import org.ethereum.beacon.core.operations.attestation.AttestationDataAndCustodyBit;
+import org.ethereum.beacon.core.spec.ChainSpec;
+import org.ethereum.beacon.validator.BeaconChainAttester;
+import org.ethereum.beacon.validator.crypto.MessageSigner;
+import tech.pegasys.artemis.ethereum.core.Hash32;
+import tech.pegasys.artemis.util.bytes.Bytes8;
+import tech.pegasys.artemis.util.bytes.Bytes96;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+import tech.pegasys.artemis.util.bytes.MutableBytesValue;
+import tech.pegasys.artemis.util.uint.UInt24;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+public class BeaconChainAttesterImpl implements BeaconChainAttester {
+
+  private SpecHelpers specHelpers;
+  private ChainSpec chainSpec;
+
+  @Override
+  public Attestation attest(
+      UInt24 index,
+      List<UInt24> committee,
+      UInt64 shard,
+      ObservableBeaconState observableState,
+      MessageSigner<Bytes96> signer) {
+    BeaconState state = observableState.getLatestSlotState();
+
+    UInt64 slot = state.getSlot();
+    Hash32 beaconBlockRoot = specHelpers.hash_tree_root(observableState.getHead());
+    Hash32 epochBoundaryRoot = getEpochBoundaryRoot(state, observableState.getHead());
+    Hash32 shardBlockRoot = Hash32.ZERO; // Note: This is a stub for phase 0.
+    Hash32 latestCrosslinkRoot = getLatestCrosslinkRoot(state, shard);
+    UInt64 justifiedSlot = state.getJustifiedSlot();
+    Hash32 justifiedBlockRoot = getJustifiedBlockRoot(state, justifiedSlot);
+    AttestationData data =
+        new AttestationData(
+            slot,
+            shard,
+            beaconBlockRoot,
+            epochBoundaryRoot,
+            shardBlockRoot,
+            latestCrosslinkRoot,
+            justifiedSlot,
+            justifiedBlockRoot);
+
+    BytesValue participationBitfield = getParticipationBitfield(index, committee);
+    BytesValue custodyBitfield = getCustodyBitfield(index, committee);
+    Bytes96 aggregateSignature = getAggregateSignature(state, data, signer);
+
+    return new Attestation(data, participationBitfield, custodyBitfield, aggregateSignature);
+  }
+
+  /*
+   Note: This can be looked up in the state using
+     get_block_root(state, head.slot - head.slot % EPOCH_LENGTH).
+  */
+  private Hash32 getEpochBoundaryRoot(BeaconState state, BeaconBlock head) {
+    UInt64 epochBoundarySlot =
+        head.getSlot().minus(head.getSlot().modulo(chainSpec.getEpochLength()));
+    return specHelpers.get_block_root(state, epochBoundarySlot);
+  }
+
+  /*
+   Set attestation_data.latest_crosslink_root = state.latest_crosslinks[shard].shard_block_root
+     where state is the beacon state at head and shard is the validator's assigned shard.
+  */
+  private Hash32 getLatestCrosslinkRoot(BeaconState state, UInt64 shard) {
+    return state.getLatestCrosslinks().get(safeInt(shard)).getShardBlockRoot();
```

Comment:
`safeInt` can be replaced with `UInt64.getIntValue()`. May be remove it from `SpecHelpers` ?

Reply:
Created corresponding issue https://github.com/harmony-dev/beacon-chain-java/issues/50.